validate.go

```diff
@@ -163,6 +163,8 @@ func checkLinux(spec rspec.Linux, rootfs string) {
 	for index := 0; index < len(spec.Namespaces); index++ {
 		if !namespaceValid(spec.Namespaces[index]) {
 			logrus.Fatalf("Namespace %s is invalid.", spec.Namespaces[index])
+		} else if spec.Namespaces[index].Type == rspec.UTSNamespace && rspec.Spec.Hostname != "" {
```

Comment:
This probably needs the whole rspec.Spec, not its current rspec.Linux.  Then you can check `spec.Hostname` and `spec.Linux.Namespaces`.  With cross-platform options like ‘hostname’ depending on platform-specific options like UTS namespaces, this kind of cross-cutting is unavoidable.